packages/xml-views-validation/test/test-utils.ts

```diff
@@ -30,3 +31,28 @@ export function testValidationsScenario(opts: {
   });
   opts.assertion(issues);
 }
+
+export function computeExpectedRanges(markedXMLSnippet: string): OffsetRange[] {
+  const expectedRanges = [];
+  const markedPattern = /🢂(.+?)🢀/gu;
+  let match;
+  // eslint-disable-next-line no-cond-assign
+  while ((match = markedPattern.exec(markedXMLSnippet))) {
+    const start = match.index;
```

Comment:
Doesn't the range-opening arrow cause an offset-by-1 issue here? For a string like `🢂a🢀` the `start` and `end` will both be in index 1 although they should be at `{0, 0}` because the actual string is `a`...

Reply:
> /🢂(.+?)🢀/gu.exec("🢂a🢀").index // -> 0

index is at the start of the match, which is before the "🢂"

Reply:
Ah, ok. I thought it was the group index.